tests/ci/integration/run_mariadb_integration.sh

```diff
@@ -0,0 +1,93 @@
+
+# The CFlags, skipped test list, and patches were taken from the internal CI pipeline of our customer.
+
+# Assumes script is executed from the root of aws-lc directory
+AWS_LC_DIR=$(pwd)
+cd ../
+ROOT=$(pwd)
+
+SCRATCH_FOLDER=${ROOT}/"MARIADB_BUILD_ROOT"
+MARIADB_SRC_FOLDER="${SCRATCH_FOLDER}/server"
+MARIADB_BUILD_FOLDER="${SCRATCH_FOLDER}/server/mariadb-aws-lc"
+MARIADB_PATCH_FOLDER=${AWS_LC_DIR}/"tests/ci/integration/mariadb_patch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${MARIADB_SRC_FOLDER}/aws-lc-install"
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+function mariadb_build() {
+  cmake ${MARIADB_SRC_FOLDER} -GNinja -DWITH_SSL=${AWS_LC_INSTALL_FOLDER} "-B${MARIADB_BUILD_FOLDER}" -DPLUGIN_COLUMNSTORE=NO -DPLUGIN_ROCKSDB=NO -DPLUGIN_S3=NO -DPLUGIN_MROONGA=NO -DPLUGIN_CONNECT=NO -DPLUGIN_SPHINX=NO -DPLUGIN_SPIDER=NO -DPLUGIN_TOKUDB=NO -DPLUGIN_PERFSCHEMA=NO -DWITH_WSREP=OFF
+  ninja -C ${MARIADB_BUILD_FOLDER}
+  ls -R ${MARIADB_BUILD_FOLDER}
+}
+
+# Used to access debugging logs.
+function dump_log() {
+  ls ${MARIADB_BUILD_FOLDER}/mysql-test/var/log
+  for logfile in $(find -L "${MARIADB_BUILD_FOLDER}/mysql-test/var/log" -type f -name '*.log'); do
+    echo "Dumping out logs to observe:"
+    cat $logfile
+  done
+}
+
+function mariadb_run_tests() {
+  pushd ${MARIADB_BUILD_FOLDER}
+  # More complicated integration tests. mtr expects to be launched in-place and with write access to it's own directories
+  #
+  # main.plugin_load passes, but is skipped over since it generates a warning when we run the script in Codebuild. Warnings will cause
+  # a failure in MariaDB's test runs, unless --nowarnings is turned on. The warning is not reproducable in Gitlab's CI or any local
+  # container runs. This test isn't relevant to AWS-LC integration so we skip over the Codebuild specific issue for now.
+  pushd mysql-test
+  echo "main.mysqldump : Field separator argument is not what is expected; check the manual when executing 'SELECT INTO OUTFILE'
+main.flush_logs_not_windows : query 'flush logs' succeeded - should have failed with error ER_CANT_CREATE_FILE (1004)
+main.mysql_upgrade_noengine : upgrade output order does not match the expected
+main.plugin_load : This test generates a warning in Codebuild. Skip over since this isn't relevant to AWS-LC. "> skiplist
+  ./mtr --suite=main --force --parallel=auto --skip-test-list=${MARIADB_BUILD_FOLDER}/mysql-test/skiplist --retry-failure=2
+  popd
+  popd
+}
+
+# TODO: Remove this when we make an upstream contribution.
+function mariadb_patch() {
+  for patchfile in $(find -L "${MARIADB_PATCH_FOLDER}" -type f -name '*.patch'); do
+    echo "Apply patch $patchfile..."
+    patch -p1 --quiet -i "$patchfile"
+  done
+}
+
+# Get latest mariadb version, we can pin to a specific version if MariaDB's code changes break us too often.
+git clone https://github.com/MariaDB/server.git ${MARIADB_SRC_FOLDER} --depth 1
```

Comment:
Since we're also applying patches, testing with the latest commit seems particularly volatile. 

I'm curious about why we [test a specific tag from mysql](https://github.com/aws/aws-lc/pull/1051) (a project that doesn't require patches), but we test the latest commit from MariaDB? 

I'm ok with either decision, it just seems we should be consistent about how we integration test.

Reply:
We've noticed that MySQL's changes have been really closely tied with OpenSSL and are continuously introducing changes that introduce new dependencies on OpenSSL symbols. Asides from that, we recently had multiple test failures due to a bug in the latest version of mySQL 2-3 months ago. The bug was resolved a few weeks ago, but it shows that we can't quite rely on the latest mySQL version as a CI target.
MariaDB has less churn and there's a better chance of us sending these patches upstream to integrate with them. I could be proven wrong, but we can easily pin a version if that's the case. MySQL actually needs patches too, but we just haven't introduced them yet since we haven't turned on testing.
* https://github.com/aws/aws-lc/pull/1063